library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/WindowInspector.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.view.View
+import android.view.inspector.WindowInspector
+import java.lang.reflect.Field
+
+@SuppressLint("PrivateApi")
+internal object WindowInspector {
+
+    private val GLOBAL_WM_CLASS by lazy {
+        try {
+            return@lazy Class.forName("android.view.WindowManagerGlobal")
```

Comment:
this will probably be obfuscated in the release build, so we need to add Proguard rules to be shipped with the library to prevent this obfuscation. Same for the reflection calls below.

Reply:
Good point

Reply:
Classes from the android framework are not embedded in the apk, but provided by the OS, so they are not obfuscated

Reply:
indeed, good point.